markdown/extensions/toc.py

```diff
@@ -256,7 +258,7 @@ def run(self, doc):
 
                 if self.use_anchors:
                     self.add_anchor(el, el.attrib["id"])
-                if self.use_permalinks:
+                if self.use_permalinks is not False:
```

Comment:
I'm not sure I'm comfortable with this change from a backward compatible perspective. It could easily break existing users stuff. As far as I can tell, this change is not necessary for the other changes here. Therefore, let's back it out.

Reply:
Upon further review. I now realize that this is necessary to allow an empty permalink, which is desirable for using icon fonts form CSS.